core/src/main/scala/kafka/server/BrokerToControllerChannelManager.scala

```diff
@@ -221,7 +216,7 @@ class BrokerToControllerRequestThread(networkClient: KafkaClient,
       } else {
         // need to backoff to avoid tight loops
         debug("No controller defined in metadata cache, retrying after backoff")
-        backoff()
```

Comment:
Hmm, this seems strange, though maybe I'm missing something.

If we get here, `activeController` is not defined. If we then pollOnce it looks like the request produced by `generateRequests` will get an exception since the activeController Option is empty. 

Previously, the backoff method was just pausing the thread for some time.

Reply:
Also, if we don't need `backoff` anymore, we can remove it since this was the only usage

Reply:
Good catch. Will fix. Let me check on test cases.

Reply:
While addressing this issue, I realized the current timeout logic does not handle the case when the controller is not known. This will cause the requests to keep piling up until we find the controller.